frontend/webclient/script/professorcontroller.js

```diff
@@ -0,0 +1,260 @@
+
+  // #########################################################################
+  // List of jobs
+
+  // all_jobs contains objects of the type {title: "title", display: function(title), students: []}
+  $scope.all_jobs = [];
+
+  // get data for specific job and student
+  $scope.get_data = function(subtype) {
+    if (subtype == 'postprocessor') {
+      $scope.reset_result_postpro();
+    } else if (subtype == 'annotated') {
+      $scope.reset_annotated_result();
+    }
+
+    console.log("Get data called with title [" + $scope.current_job_title + "], studentid [" + $scope.current_studentid + "]");
+    var msgobj = {};
+    msgobj.type = 'retrieve_result';
+    msgobj.subtype = subtype;
+    msgobj.id = $globals.token;
+    msgobj.title = $scope.current_job_title;
+    msgobj.student = $scope.current_student_id;
+    $globals.send(JSON.stringify(msgobj));
+  };
+
+  // Student results
+  $scope.reset_result_postpro = function() {
+    $scope.student_result = {};
+    $scope.student_result.letter_score = 'NA';
+    $scope.student_result.number_score = '0';
+    $scope.student_result.annotations = 'NA';
+    $scope.student_result.title = 'NA';
+    $scope.student_result.student = 'NA'
+  };
+
+
+  // Annotated files
+  $scope.reset_annotated_result = function() {
+    $scope.annotated_files = {};
+  };
+  
+  // ###################################################################
+  // Feedback management and annotation editing
+  
+  // editlist: list of old annotations that need to be changed/removed
+  $scope.editing_file = null;
+
+  $scope.editing_data_entry = null;
+  
+  $scope.editing_lineno = 0;
+  
+  $scope.feedback_sent = '';
+  
+  // The source code being used
+  // each element has lineno and text
+  $scope.editing_source = [];
+  $scope.editing_source_cache = "";
+  $scope.add_editing_source = function(lineno, text) {
+      // check if this line number already exists
+      for (var i = 0; i < $scope.editing_source.length; i++) {
+        if (lineno == $scope.editing_source[i].lineno) {
+          return;
+        }
+      }
+      
+      var source_obj = {};
+      source_obj.lineno = lineno;
+      source_obj.text = text;
+      $scope.editing_source.push(source_obj);
+      
+      // sort
+      $scope.editing_source.sort(function(a, b) {return a.lineno - b.lineno});
+      
+      var acc = '';
+      for (var i = 0; i < $scope.editing_source.length; i++) {
+          acc += $scope.editing_source[i].text;
+          if (i < $scope.editing_source.length - 1) {
+              acc += '\n';
+          }
+      }
+      $scope.editing_source_cache = acc;
+  }
+  
+  $scope.add_feedback_annotation = function(filename, lineno, text, oldannotation, data_entry) {
+      if (!$scope.editing_file) {
+          $scope.editing_data_entry = data_entry;
+          $scope.editing_file = filename;
+          $scope.editing_data_entry.editing_annotation = oldannotation;
+          $scope.editing_data_entry.editing_ann_type = "semantic";
```

Comment:
I think it's being used as a default value - probably better to have something like "Please write error type here"